split-splice/challenges-05.test.js

```diff
@@ -0,0 +1,325 @@
+'use strict';
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 1
+//
+// Write a function named howMuchPencil that takes in a string, as written on the side of a pencil.
+// As you sharpen the pencil, the string will become shorter and shorter, starting by removing the
+// first letter.
+//
+// Your function should use slice within a loop and return an array of each successive string
+// result from losing letters to the sharpener, until nothing is left.
+//
+// For example, if the input is 'Welcome', the output will be:
+// ['Welcome', 'elcome', 'lcome', 'come', 'ome', 'me', 'e', ''].
+// ------------------------------------------------------------------------------------------------
+
+const howMuchPencil = (name) => {
+  let result = [];
+  // Solution code here...
+  for(let i = 0; i < name.length; i++){
+    result.push(name.slice(i));
+  }
+  result.push('');

```

Comment:
if you change your `for` loop to stop at `i < name.length + ` then you won't need this line